.github/workflows/self-hosted.yml

```diff
@@ -74,26 +74,14 @@ jobs:
           python3 -m tools.fast_editable --path .
           python3 -m sentry.build.main
 
-      - run: docker login --username '${{ github.actor }}' --password '${{ github.token }}' ghcr.io
-        if: github.ref_name == 'master' && github.event_name != 'pull_request'
-
-      - name: Set up Docker Buildx
-        uses: docker/setup-buildx-action@b5ca514318bd6ebac0fb2aedd5d36ec1b5c232a2 # v3.10.0
-
-      - name: Build
-        uses: docker/build-push-action@32945a339266b759abcbdc89316275140b0fc960 # v6.8.10
+      - uses: getsentry/action-build-and-push-images@fce78c005f99fa7b6fed58d37187394a30d2118c
         with:
-          context: .
-          file: 'self-hosted/Dockerfile'
-          cache-from: ghcr.io/getsentry/sentry:latest
-          cache-to: type=inline
+          image_name: 'sentry'
           platforms: linux/${{ matrix.platform }}
-          tags: sentry:${{ matrix.platform }}
+          dockerfile_path: 'self-hosted/Dockerfile'
+          tag_suffix: ${{ matrix.platform }}
+          ghcr: true
           outputs: type=docker,dest=/tmp/sentry-${{ matrix.platform }}.tar
-          push: false
-          build-args: |
-            SOURCE_COMMIT=${{ github.sha }}
-            TARGETARCH=${{ matrix.platform }}
 
       - name: Upload Image
```

Comment:
i would have expected everything below here to be deleted?

Reply:
If we try and publish a multi-platform image using the composite action it takes longer due to QEMU emulation, ~7-8 minutes from the 3 minutes it takes today. I figured, this would be a good step forwards first, and we can find optimizations later on